js/components/dateRange.js

```diff
@@ -234,21 +216,20 @@ Fliplet.FormBuilder.field('dateRange', {
             end: this.formatDate(moment().subtract(1, 'days'))
           };
         default:
-          return {
-            start: this.formatDate(),
-            end: this.formatDate()
-          };
+          return this.value;
       }
     },
     formatDate: function(date) {
-      return typeof date !== 'undefined' && moment(date).isValid() ? moment(date).locale('en').format('YYYY-MM-DD') : moment(date).locale('en').format('YYYY-MM-DD');
+      return typeof date !== 'undefined' && moment(date).isValid()
+        ? moment(date).locale('en').format('YYYY-MM-DD')
+        : moment(date).locale('en').format('YYYY-MM-DD');
     },
     onBeforeSubmit: function(data) {
       // Empty date fields are validated to null before this hook is called
       if (this.autofill === 'always' && data[this.name] === null) {
         data[this.name] = this.defaultSource === 'submission'
           ? { start: '', end: '' }
```

Comment:
If it uses "submission" as the default source, should it not use the date at the point of submission, instead of an empty string?